frost-core/src/tests.rs

```diff
@@ -325,3 +333,117 @@ where
+    let serialized: <C::Group as Group>::Serialization =
+        <C::Group as Group>::Serialization::try_from(hex::decode(input).unwrap())
+            .debugless_unwrap();
+
+    let element = <C::Group as Group>::deserialize(&serialized).unwrap();
+
+    let expected = CoefficientCommitment::<C>(element);
+
+    let coeff_commitment =
+        frost::keys::CoefficientCommitment::<C>::deserialize(serialized).unwrap();
+
+    assert!(coeff_commitment.0 == expected.0);
+}
+
+/// Test error handling for creation of a coefficient commitment
+pub fn check_create_coefficient_commitment_error<C: Ciphersuite + PartialEq>(input: &str) {
+    let serialized: <C::Group as Group>::Serialization =
+        <C::Group as Group>::Serialization::try_from(hex::decode(input).unwrap())
+            .debugless_unwrap();
+
+    let coeff_commitment = frost::keys::CoefficientCommitment::<C>::deserialize(serialized);
+
+    assert!(coeff_commitment.is_err());
+}
+
+/// Test retrieving Element from CoefficientCommitment
+pub fn check_get_value_of_coefficient_commitment<C: Ciphersuite, R: RngCore + CryptoRng>(
+    mut rng: R,
+) {
+    let scalar = <<C::Group as Group>::Field>::random(&mut rng);
+    let element = <C::Group>::generator() * scalar;
+
+    let coeff_commitment = frost::keys::CoefficientCommitment::<C>(element);
+    let value = coeff_commitment.value();
+
+    assert!(value == element)
+}
+
+fn serialize_coeff_commitments<C: Ciphersuite>(
+    elements: &Vec<&str>,
+) -> Vec<<<C as Ciphersuite>::Group as Group>::Serialization> {
+    let mut coefficient_commitments = Vec::new();
+    for e in elements {
+        let serialized = <C::Group as Group>::Serialization::try_from(hex::decode(e).unwrap())
+            .debugless_unwrap();
+        coefficient_commitments.push(serialized)
+    }
+    coefficient_commitments
+}
+
+/// Test retrieving CoefficientCommitments from VerifiableSecretSharingCommitment
+pub fn check_serialize_vss_commitment<C: Ciphersuite>(commitment_helper_functions: &Value) {
+    let values = &commitment_helper_functions["elements"];
+
+    // Generate test CoefficientCommitments
+
+    // ---
+    let input_1 = values["element_1"].as_str().unwrap();
+    let input_2 = values["element_2"].as_str().unwrap();
+    let input_3 = values["element_3"].as_str().unwrap();
+
+    let serialized_coeff_commitments =
+        serialize_coeff_commitments::<C>(&vec![input_1, input_2, input_3]);
+
+    let mut coeff_commitments = Vec::new();
+    for c in serialized_coeff_commitments {
+        let coeff_commitment: CoefficientCommitment<C> =
+            CoefficientCommitment::deserialize(c).unwrap();
+        coeff_commitments.push(coeff_commitment)
+    }
+
+    //    ---
+
+    let expected = serialize_coeff_commitments::<C>(&vec![input_1, input_2, input_3]);
+
+    let vss_commitment = VerifiableSecretSharingCommitment(coeff_commitments).serialize();
+
+    assert!(expected == vss_commitment);
```

Comment:
See other comment. This checks if all elements in each vector are the same, with `as_ref()` since the `Serialization` type implements `AsRef<[u8]>`

```suggestion
    assert!(expected.len() == vss_commitment.len());
    assert!(expected
        .iter()
        .zip(vss_commitment.iter())
        .all(|(e, c)| e.as_ref() == c.as_ref()));
```

Reply:
Done [dfc0f32](https://github.com/ZcashFoundation/frost/pull/361/commits/dfc0f324a850f164934311114eff024d2e16d993)